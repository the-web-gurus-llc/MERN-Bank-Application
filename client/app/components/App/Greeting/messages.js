import { defineMessages } from 'react-intl';

export const scope = 'app.components.Greeting';

export default defineMessages({
  goodMorning: {
    id: `${scope}.good.morning`,
    defaultMessage: 'Good morning',
  },
  goodEvening: {
    id: `${scope}.good.evening`,
    defaultMessage: 'Good evening',
  },
  lastSuccessfulLogin: {
    id: `${scope}.last.successful.login`,
    defaultMessage: 'Last successful login:',
  },
  lastFailedLogin: {
    id: `${scope}.last.failed.login`,
    defaultMessage: 'Last failed login:',
  },
});
