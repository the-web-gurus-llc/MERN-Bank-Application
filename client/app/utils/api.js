const API_BASE_URL = 'https://api.pietrzakadrian.com/bank';
const AUTH = 'Auth';
const USERS = 'Users';
const CURRENCIES = 'Currencies';
const BILLS = 'Bills';
const TRANSACTIONS = 'Transactions';
const MESSAGES = 'Messages';
const NOTIFICATIONS = 'Notifications';

export default {
  auth: {
    login: `${API_BASE_URL}/${AUTH}/login`,
    register: `${API_BASE_URL}/${AUTH}/register`,
    logout: `${API_BASE_URL}/${AUTH}/logout`,
    forgetPassword: `${API_BASE_URL}/${AUTH}/password/forget`,
    resetPassword: `${API_BASE_URL}/${AUTH}/password/reset`,
  },
  users: (path = '') => {
    switch (path) {
      case 'checkEmail':
        return (email) => `${API_BASE_URL}/${USERS}/${email}/checkEmail`;
      default:
        return `${API_BASE_URL}/${USERS}`;
    }
  },
  currencies: `${API_BASE_URL}/${CURRENCIES}`,
  messages: `${API_BASE_URL}/${MESSAGES}`,
  notifications: `${API_BASE_URL}/${NOTIFICATIONS}`,
  bills: (path = '') => {
    switch (path) {
      case 'amountMoney':
        return `${API_BASE_URL}/${BILLS}/amountMoney`;
      case 'accountBalance':
        return `${API_BASE_URL}/${BILLS}/accountBalance`;
      case 'accountBalanceHistory':
        return `${API_BASE_URL}/${BILLS}/accountBalanceHistory`;
      case 'search':
        return (accountBillNumber) =>
          `${API_BASE_URL}/${BILLS}/${accountBillNumber}/search`;
      default:
        return `${API_BASE_URL}/${BILLS}`;
    }
  },
  transactions: (path = '') => {
    switch (path) {
      case 'create':
        return `${API_BASE_URL}/${TRANSACTIONS}/create`;
      case 'confirm':
        return `${API_BASE_URL}/${TRANSACTIONS}/confirm`;
      case 'authorizationKey':
        return (uuid) =>
          `${API_BASE_URL}/${TRANSACTIONS}/${uuid}/authorizationKey`;
      case 'confirmationFile':
        return (uuid, locale) =>
          `${API_BASE_URL}/${TRANSACTIONS}/${uuid}/${locale}/confirmationFile`;
      default:
        return `${API_BASE_URL}/${TRANSACTIONS}`;
    }
  },
};
