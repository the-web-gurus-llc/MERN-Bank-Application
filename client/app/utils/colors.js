export default {
  white: '#fff',
  black: '#000',
  primaryBlue: '#1890ff',
  grey: 'rgb(242, 244, 247)',
  red: 'rgb(229, 0, 0)',
  redErr: '#ff4d4f',
  silver: '#bababa',
};
