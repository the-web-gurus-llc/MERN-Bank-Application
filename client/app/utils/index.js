export { default as colors } from './colors';
export { default as media } from './media';
export { default as request } from './request';
export { default as typography } from './typography';
export { default as api } from './api';
export { default as routes } from './routes';
export { default as locales } from './locales';
export { default as i18n } from './i18n';
