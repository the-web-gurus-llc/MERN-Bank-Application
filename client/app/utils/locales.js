exports.DEFAULT_LOCALE = 'en';
exports.DEFAULT_DATE_FORMAT = 'dd.MM.yyyy, hh:mm aa';

// prettier-ignore
exports.appLocales = [
  'en',
  'de',
  'pl',
];
