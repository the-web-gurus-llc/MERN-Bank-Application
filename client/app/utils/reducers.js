/**
 * Combine all reducers in this file and export the combined reducers.
 */

import { combineReducers } from 'redux';
import { connectRouter } from 'connected-react-router';

import history from 'utils/history';
import globalReducer from 'containers/App/reducer';
import languageProviderReducer from 'providers/LanguageProvider/reducer';
import loadingProviderReducer from 'providers/LoadingProvider/reducer';
import errorProviderReducer from 'providers/ErrorProvider/reducer';

/**
 * Merges the main reducer with the router state and dynamically injected reducers
 */
export default function createReducer(injectedReducers = {}) {
  const rootReducer = combineReducers({
    global: globalReducer,
    language: languageProviderReducer,
    loading: loadingProviderReducer,
    error: errorProviderReducer,
    router: connectRouter(history),
    ...injectedReducers,
  });

  return rootReducer;
}
