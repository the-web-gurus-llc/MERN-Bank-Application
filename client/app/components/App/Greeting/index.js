import React from 'react';
import { createStructuredSelector } from 'reselect';
import { makeSelectUser } from 'containers/App/selectors';
import { useSelector } from 'react-redux';
import { format, getHours } from 'date-fns';
import { makeSelectDateFormat } from 'providers/LanguageProvider/selectors';
import { FormattedMessage } from 'react-intl';
import {
  StyledGreeting,
  StyledNameWrapper,
  StyledLastSuccessfulLoggedDateWrapper,
  StyledLastFailedLoggedDate,
} from './styles';
import messages from './messages';

const stateSelector = createStructuredSelector({
  user: makeSelectUser(),
  dateFormat: makeSelectDateFormat(),
});

export default function Greeting() {
  const { user, dateFormat } = useSelector(stateSelector);

  return (
    <StyledGreeting>
      <div>
        {getHours(new Date()) > 20 ? (
          <FormattedMessage {...messages.goodEvening} />
        ) : (
          <FormattedMessage {...messages.goodMorning} />
        )}
        ,{' '}
        <StyledNameWrapper>
          {user?.firstName} {user?.lastName}
        </StyledNameWrapper>
      </div>

      {user?.userAuth?.lastSuccessfulLoggedDate && (
        <div>
          <FormattedMessage {...messages.lastSuccessfulLogin} />{' '}
          <StyledLastSuccessfulLoggedDateWrapper>
            {format(
              new Date(user?.userAuth?.lastSuccessfulLoggedDate),
              dateFormat,
            )}
          </StyledLastSuccessfulLoggedDateWrapper>
        </div>
      )}

      {user?.userAuth?.lastFailedLoggedDate && (
        <div>
          <FormattedMessage {...messages.lastFailedLogin} />{' '}
          <StyledLastFailedLoggedDate>
            {format(new Date(user?.userAuth?.lastFailedLoggedDate), dateFormat)}
          </StyledLastFailedLoggedDate>
        </div>
      )}
    </StyledGreeting>
  );
}
