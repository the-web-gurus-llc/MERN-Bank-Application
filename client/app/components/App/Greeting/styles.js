import styled from 'styled-components';
import { colors, typography } from 'utils';

export const StyledGreeting = styled.section`
  text-align: right;
  margin: 0 0 20px;
  line-height: 1.3;
`;

export const StyledNameWrapper = styled.span`
  color: ${colors.primaryBlue};
  font-weight: ${typography.fontWeightBold};
`;

export const StyledLastSuccessfulLoggedDateWrapper = styled.span`
  color: ${colors.black};
`;

export const StyledLastFailedLoggedDate = styled.span`
  color: ${colors.red};
`;
