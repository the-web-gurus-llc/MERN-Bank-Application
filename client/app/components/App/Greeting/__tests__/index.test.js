import React from 'react';
import { render } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { Provider } from 'react-redux';
import { IntlProvider } from 'react-intl';
import { DEFAULT_LOCALE } from 'utils/locales';
import configureStore from 'utils/configureStore';
import Greeting from '../index';

describe('<Greeting />', () => {
  const history = createMemoryHistory();
  const store = configureStore({}, history);

  it('should render a Greeting', () => {
    const { container } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <Greeting />
        </IntlProvider>
      </Provider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
