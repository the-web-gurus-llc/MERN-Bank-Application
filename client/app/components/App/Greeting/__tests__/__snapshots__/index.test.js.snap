// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Greeting /> should render a Greeting 1`] = `
<section
  class="Greetingstyle__StyledGreeting-r6yczo-0 gpylKs"
>
  <div>
    <span>
      Good morning
    </span>
    ,
     
    <span
      class="Greetingstyle__StyledNameWrapper-r6yczo-1 ifLnoL"
    >
       
    </span>
  </div>
</section>
`;
